File: javascript/342.power-of-four.js
/*
 * @lc app=leetcode id=342 lang=javascript
 *
 * [342] Power of Four
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function (n) {
  if (n > 0) {
    while (n % 4 === 0) {
      n /= 4;
    }
    return n === 1;
  }
  return false;
};
// @lc code=end

File: rust/342.power-of-four.rs
/*
 * @lc app=leetcode id=342 lang=rust
 *
 * [342] Power of Four
 */

// @lc code=start
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n > 0 {
            let mut n = n;
            while n % 4 == 0 {
                n /= 4;
            }
            return n == 1;
        }
        return false;
    }
}
// @lc code=end
